src/ape/api/projects.py

```diff
@@ -146,8 +146,11 @@ def _create_compilers_list(
                 version = versions[0]
                 filtered_paths = [p for p in source_paths if p.suffix == ext]
                 version_map = {version: filtered_paths}
+            
+            settings = compiler.get_compiler_settings(source_paths,contracts_folder)
 
             for version, paths in version_map.items():
+                settings = compiler.get_compiler_settings(paths,contracts_folder)
```

Comment:
We are already getting all the settings on line 150, this is a duplicate line.
Instead, grab the settings that pertains to this version, something like:

```python
version_settings = settings[version]
```